### PR TITLE
State: remove Lodash stub functions and withoutPersistence from several reducers

### PR DESCRIPTION
**Pull request by jsnajdr**
Removes `stubFalse` and `stubTrue` usages from several simple reducers. Also removes some `withoutPersistence` usages within `keyedReducer`. No persistence is the default and the `keyedReducer` would have to declare `.hasCustomPersistence = true` flag to enable it.

### CODE REVIEW

**Comment by jsnajdr** [comment]
OK, this PR escalated quickly 😆 

I had to fixup the affected unit test because of a slight behavior change. When a `keyedReducer` changes sub-state to the initial value, the record is removed completely. Some states then transition from
```js
{ 123: true, 456: true }
```
not to
```js
{ 123: true, 456: false }
```
but to
```js
{ 123: true }
```
The default `false` value was removed. Selectors take care of that because they are implemented as:
```js
const isEnabled( state, siteId ) => state[ siteId ] ?? false;
```

I modified the tests to use a mock selector like this. Then the tested behavior doesn't depend on the exact value of the state.

I also removed the `state.jetpack.connection.disconnectRequests` reducer and all associated actions, because I discovered they are not used anywhere.

File: client/state/jetpack/connection/reducer.js [hunk 1]
```diff
@@ -98,42 +75,21 @@ export const dataItems = keyedReducer(
...
+export const dataRequests = keyedReducer( 'siteId', ( state = false, action ) => {
+	switch ( action.type ) {
+		case JETPACK_USER_CONNECTION_DATA_REQUEST:
+			return true;
+		case JETPACK_USER_CONNECTION_DATA_REQUEST_FAILURE:
+		case JETPACK_USER_CONNECTION_DATA_REQUEST_SUCCESS:
+			return false;
+	}
 
-export const disconnectRequests = keyedReducer(
```
Comment (tyxla) [inline client/state/jetpack/connection/reducer.js:117]: Nice cleanup 👍

**Review by tyxla — APPROVED** [review]
This looks great 🚢

**Comment by jsnajdr** [comment]
@tyxla Thanks for reviewing. After testing the UI by disconnecting a real Jetpack site, I added one more patch that removes a noop `catch` handler in 7a37a9c.

The action is used here: https://github.com/Automattic/wp-calypso/blob/trunk/client/blocks/disconnect-jetpack/index.jsx#L186-L202

And it uses the promise returned from `dispatch` to show a success or error notice. Without removing the empty `catch` block, it would show success notice even if the disconnection failed.

Please have a quick look before I merge this.

**Review by tyxla — APPROVED** [review]
Still looks good 🚢  Thanks for the disconnection fix!

**Pull request merged**